aten/src/ATen/native/Sorting.cpp

```diff
@@ -216,6 +217,10 @@ std::tuple<Tensor&, Tensor&> kthvalue_out_impl_cpu(
   // FIXME: This seems bogus, I only do this because it was the old behaviour.
   //        The reductions are fine, as long as the axis being reduced along
   //        isn't of 0 elements (and the output has elements).
+  auto lap = at::get_overlap_status(self, values);
+  TORCH_CHECK(lap != at::MemOverlapStatus::PARTIAL &&
```

Comment:
Perhaps it would be good to move this logic to a new function `void assert_no_overlap(const Tensor& a, const Tensor& b)` in `MemoryOverlap.h/cpp`, since we'll need to perform this check in at least one other place (cuda)

Reply:
Done!